drivers/storage/portworx/component/monitoring.go

```diff
@@ -99,6 +157,10 @@ func (c *monitoring) Delete(cluster *corev1.StorageCluster) error {
 	if err != nil && !metaerrors.IsNoMatchError(err) {
 		return err
 	}
+	err = k8sutil.DeleteDeployment(c.k8sClient, GrafanaDeploymentName, cluster.Namespace, *ownerRef)
+	if err != nil && !metaerrors.IsNoMatchError(err) {
```

Comment:
nit: Don't have to check for the `metaerrors.IsNoMatchError` error as we are just deleting a deployment and not a custom resource.

Reply:
Removed this